xray/segment.go

```diff
@@ -357,21 +357,22 @@ func (seg *Segment) Close(err error) {
 		seg.addError(err)
 	}
 
+	cancelSegCtx := seg.cancelCtx
```

Comment:
another way: don't create routine for Sampled segments

```
	// create a new context to close it on segment close;
	// this makes sure the closing of the segment won't affect the client's code, that uses the returned context
	if !seg.Dummy {
		ctx1, cancelCtx := context.WithCancel(ctx)
		seg.cancelCtx = cancelCtx
		go func() {
			<-ctx1.Done()
			seg.handleContextDone()
		}()
	}
```

Reply:
For me it's better because you don't send sampled segments and don't create extra go routines

Reply:
Hi @NathanielRN [here PR](https://github.com/aws/aws-xray-sdk-go/pull/365)

Reply:
Thanks @nikolaianohyn ! I think your PR is better so I will close this PR.